tests/test_variant_table.py

```diff
@@ -1530,6 +1336,71 @@ def test_masked(self):
                     np.testing.assert_equal(index_masked.format('GT'),
                                             [[VCFTable.GT_HOMOZYGOUS_REF], [VCFTable.GT_HOMOZYGOUS_ALT]])
 
+    def test_phased(self):
+        dumptext(
+            self.tmpvcf, vcf_header1, """
+            #CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	samp1
+            1	100	.	AGC	A	.	.	AF=0.1	GT	.
+            1	102	.	AGC	A	.	.	AF=0.2	GT	0/0
+            1	104	.	AGC	A	.	.	AF=0.3	GT	1/0
+            1	106	.	AGC	A	.	.	AF=0.3	GT	0/1
+            1	108	.	AGC	A	.	.	AF=0.4	GT	1/1
+            1	110	.	AGC	A	.	.	AF=0.5	GT	0|0
+            1	112	.	AGC	A	.	.	AF=0.3	GT	1|0
+            1	114	.	AGC	A	.	.	AF=0.3	GT	0|1
+            1	116	.	AGC	A	.	.	AF=0.4	GT	1|1
+            """)
+
+        genome = MiniGenome("hg19")
+
+        VCFTable.build_vcfbin(self.tmpbin, self.tmpvcf, genome, info_ids=["AF"], fmt_ids=["GT"], validate=False)
+        with VCFTable(self.tmpbin) as vt:
+            variants = vt.find_within(genome.interval('chr1', '+', 100, 120))
+            self.assertEqual(VCFTable.GT_UNKNOWN, variants[0].GT)
+            self.assertEqual(VCFTable.GT_HOMOZYGOUS_REF, variants[1].GT)
+            self.assertEqual(VCFTable.GT_HETEROZYGOUS_UNPHASED, variants[2].GT)
+            self.assertEqual(VCFTable.GT_HETEROZYGOUS_UNPHASED, variants[3].GT)
+            self.assertEqual(VCFTable.GT_HOMOZYGOUS_ALT, variants[4].GT)
+            self.assertEqual(VCFTable.GT_HOMOZYGOUS_REF, variants[5].GT)
+            self.assertEqual(VCFTable.GT_HETEROZYGOUS_PHASED_1_0, variants[6].GT)
+            self.assertEqual(VCFTable.GT_HETEROZYGOUS_PHASED_0_1, variants[7].GT)
+            self.assertEqual(VCFTable.GT_HOMOZYGOUS_ALT, variants[8].GT)
+
+
+    def test_phased_unknown(self):
+        dumptext(
+            self.tmpvcf, vcf_header1, """
+            #CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	samp1
+            1	102	.	AGC	A	.	.	AF=0.2	GT	0/.
+            1	104	.	AGC	A	.	.	AF=0.3	GT	1/.
+            1	106	.	AGC	A	.	.	AF=0.4	GT	./0
+            1	108	.	AGC	A	.	.	AF=0.5	GT	./1
+            1	112	.	AGC	A	.	.	AF=0.7	GT	0|.
+            1	114	.	AGC	A	.	.	AF=0.8	GT	1|.
+            1	116	.	AGC	A	.	.	AF=0.9	GT	.|0
+            1	118	.	AGC	A	.	.	AF=0.1	GT	.|1
```

Comment:
Just to confirm: when GT is one of `0|.`, `.|0`, `1|.`, `.|1` we're setting it to GT_UNKNOWN. Can user still extract the variant where info is available (the one that's 0 or 1)? e.g. one might want to reconstruct one of the two haplotypes.

Reply:
> Can user still extract the variant where info is available

Not with the current implementation. Are you thinking we should add more enum values, or to make the field also available as a string? We can do the latter in a later PR to avoid delaying releasing this feature, w/o breaking backwards compatibility.

Reply:
Thanks for the details! I don't have any concrete use cases right now. Will reach out again once we have a few example VCFs and have more clarity on what we'd like to do downstream of GK. Thank you!